src/manybody_gal.F

```diff
@@ -339,8 +353,15 @@ FUNCTION angular(gal, r_last_update_pbc, iparticle, cell, particle_set, nvec, en
 
          ! For fit purpose
          IF (gal%express .AND. energy) THEN
-            WRITE (*, *) "Fourier", costheta, COS(2.0_dp*theta), COS(3.0_dp*theta), &
+            logger => cp_get_default_logger()
+            index_outfile = cp_print_key_unit_nr(logger, mm_section, &
```

Comment:
Passing a null pointer for the section to `cp_print_key_unit_nr` will not work because of [this assert](https://github.com/cp2k/cp2k/blob/master/src/input/cp_output_handling.F#L902). This is currently not caught by the regtests, because `gal%express` is always false. A simple solution would be to pass in the output unit `index_outfile` instead of the section. Furthermore, it would be good to have some tests covering those branches.

Reply:
Here, I do not think it is a problem because "energy" is never TRUE while mm_section is nullified (that is actually the point of this "energy" boolean: to avoid the printing when this function is called from the force evaluation routine instead of the energy evaluation routine). So when the function is called from the energy routine, energy is TRUE and mm_section is a real section, while when it is called from the force evaluation, energy is FALSE and mm_section is nullified but cp_print_key_unit_nr is never used. I checked on my computer and it works indeed fine even with gal%express being TRUE.
About this testing, this option is only intended to print some quantities needed for a linear fit of the forcefield so I didn't consider it was worth regtesting because it would only be useful for a person that would want to redevelop the forcefield.

Reply:
Ok